### PR TITLE
Upgrade HFE to 5.0.0 to support platform 2.3+

### PR DESCRIPTION
**Pull request by ibacher**
## Description of what I changed

This refactors HFE to depend on core 2.3.2, moves the stuff from the api-2.3 / api-2.4 submodule back into the main API.

## Issue I worked on

related to https://issues.openmrs.org/browse/HTML-788

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api-tests/pom.xml [hunk 1]
```diff
@@ -150,12 +150,23 @@
...
+				</dependency>
+			</dependencies>
+		</profile>
 		<profile>
 			<id>2.4</id>
 			<dependencies>
 				<dependency>
 					<groupId>${project.groupId}</groupId>
-					<artifactId>${project.parent.artifactId}-api-2.4</artifactId>
+					<artifactId>${project.parent.artifactId}-api-2.3</artifactId>
```
Comment (mseaton) [inline api-tests/pom.xml:169]: @ibacher do we still need an api-2.3 artifact?  Or can we just have "api" now?

**Review by mogoodrich — APPROVED** [review]
I added a +1 to @mseaton 's question, but generally LGTM @ibacher .

I do think it would be good for @icrc-loliveira  to take a look, so I will add him as a reviewer!

Thanks!

File: api-tests/pom.xml (hunk 1 above)
Comment (ibacher) [reply api-tests/pom.xml:169]: Absolutely!

**Comment by ibacher** [comment]
@icrc-loliveira Are you able to at least check this out locally and confirm whether this breaks things before I merge it in? You should be able to checkout the branch locally using

```
git fetch origin pull/266/head:5.0.0-SNAPSHOT
git checkout 5.0.0-SNAPSHOT
```

Alternatively, if you have the GitHub CLI:

```
gh pr checkout 266
```

**Review by mseaton — APPROVED** [review]
I am assuming the main changes here have already been reviewed and tested in previous iterations.

**Pull request merged**